library/src/main/java/com/google/android/exoplayer2/metadata/id3/Id3Decoder.java

```diff
@@ -444,6 +450,112 @@ private static TextInformationFrame decodeTextInformationFrame(ParsableByteArray
     return new TextInformationFrame(id, description);
   }
 
+  private static UrlLinkFrame decodeUrlLinkFrame(ParsableByteArray id3Data,
+      int frameSize) throws UnsupportedEncodingException {
+    int encoding = id3Data.readUnsignedByte();
+    String charset = getCharsetName(encoding);
+
+    byte[] data = new byte[frameSize - 1];
+    id3Data.readBytes(data, 0, frameSize - 1);
+
+    int descriptionEndIndex = indexOfEos(data, 0, encoding);
+    String description = new String(data, 0, descriptionEndIndex, charset);
+
+    int urlStartIndex = descriptionEndIndex + 1;
```

Comment:
Is this correct, or should it be `+ delimiterLength(encoding)`? This also fails in the case that the url string has length 0, because there's no requirement for a terminating byte. See decodeTxxxFrame for the type of check you should add.